interfaces/repo.go

```diff
@@ -743,6 +758,11 @@ func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string) []*Slo
 // isAutoConnectCandidate returns true if the plug is a candidate to
 // automatically connect to the given slot.
 func isAutoConnectCandidate(plug *Plug, slot *Slot) bool {
+	// check if this makes sense at all
+	if slot.Interface != plug.Interface {
+		return false
+	}
```

Comment:
I didn't write the content interface, but looking at it, I think this check is ok to move here.